### PR TITLE
fix(java): ensure a corpus is always set in extracted compilation units

### PR DESCRIPTION
**Pull request by jaysachs**
Minimize churn by keeping the interface to FileVNames unchanged. Factor out an EnvironmentUtils class in which to house access to system properties / environment variables.

### CODE REVIEW

File: kythe/java/com/google/devtools/kythe/extractors/shared/EnvironmentUtils.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      return Optional.empty();
+    }
+    return Optional.of(result);
+  }
+
+  public static String defaultCorpus() {
+    return readEnvironmentVariable("KYTHE_CORPUS", DEFAULT_CORPUS);
+  }
+
+  public static final String DEFAULT_CORPUS = "kythe";
```
Comment (shahms) [inline kythe/java/com/google/devtools/kythe/extractors/shared/EnvironmentUtils.java:61]: Should we log a warning for this?

File: kythe/java/com/google/devtools/kythe/extractors/shared/ExtractorUtils.java [hunk 2]
```diff
@@ -197,4 +199,40 @@ public static CompilationUnit normalizeCompilationUnit(CompilationUnit existingC
     existingCompilationUnit = builder.build();
     return existingCompilationUnit;
   }
+
+
+  public static String readEnvironmentVariable(String variableName) {
+    return readEnvironmentVariable(variableName, null);
+  }
```
Comment (shahms) [inline kythe/java/com/google/devtools/kythe/extractors/shared/ExtractorUtils.java:206]: It looks like these are just a copy of EnvironmentUtils?

Comment (jaysachs) [reply kythe/java/com/google/devtools/kythe/extractors/shared/ExtractorUtils.java]: Removed.

File: kythe/java/com/google/devtools/kythe/extractors/shared/EnvironmentUtils.java (hunk 1 above)
Comment (jaysachs) [reply kythe/java/com/google/devtools/kythe/extractors/shared/EnvironmentUtils.java:75]: Done.

**Review by shahms — APPROVED** [review]

**Pull request merged**